### PR TITLE
feat(ui5-rating-indicator): add required and accessibleNameRef properties

### PR DESCRIPTION
**Pull request by Okiana**

### CODE REVIEW

File: packages/main/test/pages/RatingIndicator.html [hunk 1]
```diff
@@ -68,6 +68,20 @@ <h3>RatingIndicator with title</h3>
...
 
+	<h3>RatingIndicator with required</h3>
+	<ui5-label id="label-required" for="rating-indicator-required">Required label</ui5-label>
+	<ui5-rating-indicator id="rating-indicator-required" required="true"></ui5-rating-indicator>
+	<br>
+	<br>
+	<br>
+
+	<h3>RatingIndicator with accessibleNameRef</h3>
+	<ui5-label id="label-acc-name-ref" for="rating-indicator-acc-name-ref">Some ACC label</ui5-label>
```
Comment (elenastoyanovaa) [inline packages/main/test/pages/RatingIndicator.html:79]: can you add an example for accessibleName as well?

File: packages/main/src/RatingIndicator.ts [hunk 2]
```diff
@@ -161,6 +162,30 @@ class RatingIndicator extends UI5Element {
...
 	accessibleName!: string;
 
+	/**
+	 * Receives id(or many ids) of the elements that label the component.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.accessibleNameRef
+	 * @defaultvalue ""
+	 * @public
+	 * @since 1.13.0
```
Comment (elenastoyanovaa) [inline packages/main/src/RatingIndicator.ts:172]: 1.15.0 ?

File: packages/main/src/RatingIndicator.ts [hunk 3]
```diff
@@ -161,6 +162,30 @@ class RatingIndicator extends UI5Element {
...
+	 accessibleNameRef!: string;
+
+	 /**
+	 * Defines whether the component is required.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.RatingIndicator.prototype.required
+	 * @defaultvalue false
+	 * @public
+	 * @since 1.13.0
```
Comment (elenastoyanovaa) [inline packages/main/src/RatingIndicator.ts:184]: 1.15.0

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]
Please update the required logic as discussed internally and also I would expect a sample in storybook as well showcasing the accessibility. Label mapped with accessibleNameRef + a required RatingIndicator

File: packages/main/src/RatingIndicator.ts (hunk 2 above)
Comment (Okiana) [reply packages/main/src/RatingIndicator.ts:172]: done

File: packages/main/src/RatingIndicator.ts (hunk 3 above)
Comment (Okiana) [reply packages/main/src/RatingIndicator.ts:184]: Done

File: packages/main/test/pages/RatingIndicator.html (hunk 1 above)
Comment (Okiana) [reply packages/main/test/pages/RatingIndicator.html:79]: done

**Comment by Okiana** [comment]
> Please update the required logic as discussed internally and also I would expect a sample in storybook as well showcasing the accessibility. Label mapped with accessibleNameRef + a required RatingIndicator

The logic for aria description is updated and I've added a sample in the storybook

File: packages/main/src/RatingIndicator.ts [hunk 4]
```diff
@@ -305,6 +330,14 @@ class RatingIndicator extends UI5Element {
 		return this.disabled || undefined;
 	}
 
+	get _ariaLabel() {
+		return getEffectiveAriaLabelText(this);
+	}
+
+	get _ariaDescription() {
+		return this.required ? "required" : "";
```
Comment (elenastoyanovaa) [inline packages/main/src/RatingIndicator.ts:338]: the text should be translated

File: packages/main/test/specs/RatingIndicator.spec.js [hunk 5]
```diff
@@ -122,4 +122,18 @@ describe("Rating Indicator general interaction", () => {
 
 		assert.strictEqual(await ratingIndicator.getAttribute("title"), TOOLTIP, "The title attribute is rendered in the inner div as well.");
 	});
+
+	it("Tests ACC attrs - required property add aria-description", async () => {
+		const ratingIndicator = await browser.$("#rating-indicator-required");
+		const REQUIRED = "required";
```
Comment (elenastoyanovaa) [inline packages/main/test/specs/RatingIndicator.spec.js:128]: as the text should be translated you should test against the resourse bundle text (see other tests with translations)

File: packages/playground/_stories/main/RatingIndicator/RatingIndicator.stories.ts [hunk 6]
```diff
@@ -58,3 +62,18 @@ ReadonlyRatingIndicator.args = {
 	max: 7,
 	readonly: true
 };
+
+export const RequiredRatingAccessibleNameRef: StoryFn = Template.bind({});
```
Comment (elenastoyanovaa) [inline packages/playground/_stories/main/RatingIndicator/RatingIndicator.stories.ts:66]: Can you change the title of the sample?

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]

File: packages/playground/_stories/main/RatingIndicator/RatingIndicator.stories.ts (hunk 6 above)
Comment (Okiana) [reply packages/playground/_stories/main/RatingIndicator/RatingIndicator.stories.ts:66]: Done

File: packages/main/test/specs/RatingIndicator.spec.js (hunk 5 above)
Comment (Okiana) [reply packages/main/test/specs/RatingIndicator.spec.js]: Done

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]

**Comment by Okiana** [comment]
Closed due to github merge issues

**Pull request closed without merging**